integrations/gitlab/gitlab_integration/gitlab_service.py

```diff
@@ -186,6 +186,35 @@ def create_webhooks(self) -> list[int | str]:
 
         return webhook_ids
 
+    def create_system_hook(self) -> None:
+        logger.debug("Checking if system hook already exists")
+        try:
+            for hook in self.gitlab_client.hooks.list(iterator=True):
+                if hook.url == f"{self.app_host}/integration/system/hook":
+                    logger.debug("System hook already exists, no need to create")
+                    return
+        except Exception:
+            logger.error(
+                "Failed to check if system hook exists, skipping trying to create, to avoid duplicates"
```

Comment:
dont forget to log the exception
maybe you should use logger.exception

Reply:
logger.error, prints the error out of the box when inside a except :)